base/steer/FairRunSim.cxx

```diff
@@ -296,10 +296,12 @@ void FairRunSim::SetMCConfig()
 
     fApp->InitMC("foo", "bar");
 
-    if (fUseSimSetupFunction) {
+    if (fUseSimSetupPostInitFunction) {
         fSimSetupPostInit();
     } else {
```

Comment:
```suggestion
    if (fSimSetupPostInit) {
        fSimSetupPostInit();
    } else {
```

It looks to me that the bool flag is even redundant, see https://en.cppreference.com/w/cpp/utility/functional/function/operator_bool.

Also, e7dfa6a3a363d745027cfd6570ec51f2ec0af3a8 added new fields to the `FairRunSim` class, but did not bump the `ClassDef(FairRunSim, 2);` revision?!

@fuhlig1 @karabowi How do you want to proceed?

Reply:
I prefer to leave the flags for the user to set.
This is the common scheme throughout the whole code - the users are used to having a flag to set.

Reply:
I agree with Radek but thanks for pointing out the possibility to avoid the extra flags. I think this is something we should keep in mind.

Reply:
Leave the flags if you really want to.
But don't use them in new code. It makes no sense. Use what Dennis suggested.
Ideally mark the flags as deprecated.